### PR TITLE
4.next component model alias

### PR DESCRIPTION
**Pull request by dereuromark**
Alternative to https://github.com/cakephp/cakephp/pull/14351 for 4.next

### CODE REVIEW

File: src/Datasource/ModelAwareTrait.php [hunk 1]
```diff
@@ -72,6 +72,41 @@ protected function _setModelClass(string $name): void
         }
     }
 
+    /**
+     * @param string|null $modelClass Name of model class. Defaults to $this->modelClass.
```
Comment (markstory) [inline src/Datasource/ModelAwareTrait.php]:
```suggestion
    /**
     * Get the current model alias based on modelClass property.
     *
     * @param string|null $modelClass Name of model class. Defaults to $this->modelClass.
```

**Comment by ADmad** [comment]
Is the new method really necessary? Currently one can use `$controller->loadModel()->getAlias()` to get the default model's alias.

**Comment by dereuromark** [comment]
That could also be enough.

**Comment by ADmad** [comment]
Then it would be better to avoid increasing the API surface.

**Comment by dereuromark** [comment]
This also reduces side efects inside methods as it reduces complexity and delegates those into smaller chunks.
But I can see that the benefit here is really small - as such I would be fine with the status quo.

**Comment by dereuromark** [comment]
There is one problem now though

The core states:
```php
    /**
     * ...
     *
     * Use empty string to not use auto-loading on this object. Null auto-detects based on
     * controller name.
     *
     * @var string|null
     */
    protected $modelClass;
```

And then tools like bake are forced to use `loadModel()->...` as they cannot know beforehand that here no model is supposed to be used.

Causing errors like

> [UnexpectedValueException] Default modelClass is empty in /.../vendor/cakephp/cakephp/src/Datasource/ModelAwareTrait.php on line 98

Sure, one could use try catch here, but a dedicated method might be cleaner before any loadModel class is done.

**Comment by ADmad** [comment]
> And then tools like bake are forced to use loadModel()->... as they cannot know beforehand that here no model is supposed to be used.

Can you give a specific example/use case? Baking works just fine now

**Comment by dereuromark** [comment]
Actually, it doesnt. I can make a fix

**Pull request closed without merging**